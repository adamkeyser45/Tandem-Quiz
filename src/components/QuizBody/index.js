import React, { useState } from 'react';
import $ from 'jquery';

function QuizBody() {

    const questionsArray = require('../../assets/Apprentice_TandemFor400_Data.json');

    const [gameOff, setGameOff] = useState(true);
    const [tenQuestions, setTenQuestions] = useState([]);
    const [currentQuestion, setCurrentQuestion] = useState(0);
    const [score, setScore] = useState(0);
    
    function startQuiz() {
        
        setGameOff(false);
        const chosen = [];

        // choose ten random questions
        while (tenQuestions.length < 10) {
            let num = Math.floor(Math.random() * (questionsArray.length));
            
            if (chosen.includes(num)) {
                console.log(num);
            } else {
                chosen.push(num);
                tenQuestions.push(questionsArray[num]);
            }
        }

        console.log(tenQuestions);
    };

    function resetQuiz() {
        setGameOff(true);
        setTenQuestions([]);
    };

    function handleAnswerChoice () {
        const nextQuestion = currentQuestion + 1;
        if (nextQuestion < tenQuestions.length) {
            setCurrentQuestion(nextQuestion);
        } else {
           resetQuiz();
        }
        
    };

    return (
        <div className="contrainer-fluid" >
            <div className='card border m-4' style={{height: '425px'}}>
                <div className='card-header text-center'>
                    <div className='btn-group'>
                        <button className='btn btn-light border' id='startBtn' onClick={startQuiz}>
                            Start!
                        </button>
                        <div className="dropdown">
                            <button className="btn btn-light border dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                                Options
                            </button>
                            <div className="dropdown-menu" aria-labelledby="dropdownMenuButton">
                                <button type='button' className='btn btn-link' data-toggle='modal' data-target='#rulesModal'>How To Play</button>
                                <button type='button' className='btn btn-link' onClick={resetQuiz}>Reset Quiz</button>
                            </div>
                        </div>
                    </div>
                </div>

                <div className="modal fade" id="rulesModal" tabIndex="-1" role="dialog" aria-labelledby="rulesModalLabel" aria-hidden="true">
                    <div className="modal-dialog" role="document">
                        <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title" id="rulesModalLabel">How to Play!</h5>
                            <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                            <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <div className="modal-body">
                            Rules
                            <ol>
                                <li>Press Start to start the quiz</li>
                                <li>Press the button that has the correct answer for each question</li>
                                <li>When all of the questions are done, see your score!</li>
                            </ol>
                        </div>
                        </div>
                    </div>
                </div>
                
                <div className='card-body text-center'>
                    {gameOff ? (
                        <>
                        <h2>Press Start to begin the Trivia Quiz!</h2>
                        <h4>Your last score was {score}.</h4>
                        <h4>Can you do better?</h4>
                        </>
                    ) : (
                        <>
                        <p id='question' className="card-text text-center h5 border rounded p-2">{tenQuestions[currentQuestion].question}</p>
                        <div className='container pt-2'>
                            {tenQuestions[currentQuestion].choices.map((choice) => (
                                <button className="btn btn-primary btn-block m-1" onClick={handleAnswerChoice}>{choice.answer}</button>
                            ))}
                        </div>
                        </>
                    )} 
                </div>
            </div>
        </div>
    )
}

export default QuizBody;