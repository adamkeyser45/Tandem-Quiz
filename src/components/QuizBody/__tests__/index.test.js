import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import QuizBody from '..';

afterEach(cleanup);

describe('QuizBody Component', () => {
    it('renders', () => {
        render(<QuizBody />);
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<QuizBody />);
        expect(asFragment()).toMatchSnapshot();
    });
});