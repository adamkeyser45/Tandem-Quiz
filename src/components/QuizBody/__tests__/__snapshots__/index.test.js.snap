// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizBody Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="contrainer-fluid"
  >
    <div
      class="card border m-4"
      style="height: 375px;"
    >
      <div
        class="card-header text-center"
      >
        <div
          class="btn-group"
        >
          <button
            class="btn btn-light border"
            id="startBtn"
          >
            Start!
          </button>
          <div
            class="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="btn btn-light border dropdown-toggle"
              data-toggle="dropdown"
              id="dropdownMenuButton"
              type="button"
            >
              Options
            </button>
            <div
              aria-labelledby="dropdownMenuButton"
              class="dropdown-menu"
            >
              <button
                class="btn btn-link"
                data-target="#rulesModal"
                data-toggle="modal"
                type="button"
              >
                How To Play
              </button>
              <button
                class="btn btn-link"
                type="button"
              >
                Reset Quiz
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="rulesModalLabel"
        class="modal fade"
        id="rulesModal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h5
                class="modal-title"
                id="rulesModalLabel"
              >
                How to Play!
              </h5>
              <button
                aria-label="Close"
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  Ã—
                </span>
              </button>
            </div>
            <div
              class="modal-body"
            >
              <ol>
                <li>
                  Press "Start!" to begin the quiz.
                </li>
                <li>
                  Each quiz is 10 questions, chosen randomly from a list of 21, and each question will only be shown once.
                </li>
                <li>
                  Read the Trivia Question and choose one of the four answer choices.
                </li>
                <li>
                  If your answer is correct, the buttons will flash green, if it is incorrect, they will flash red.
                </li>
                <li>
                  When all of the questions are done, you should see your score. You get 10 points for each correct answer.
                </li>
                <li>
                  If you ever need to reset the quiz, click "Reset Quiz" in the "Options" menu. 
                </li>
              </ol>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card-body text-center"
      >
        <h2>
          Press Start to begin the Trivia Quiz!
        </h2>
        <h4>
          Your last score was 0.
        </h4>
        <h4>
          Can you do better?
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
